pdata/internal/json/stream.go

```diff
@@ -0,0 +1,71 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package json // import "go.opentelemetry.io/collector/pdata/internal/json"
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"strconv"
+
+	jsoniter "github.com/json-iterator/go"
+)
+
+// Stream avoids the need to explicitly call the `Stream.WriteMore` method while marshaling objects by
+// checking if a field was previously written inside the current object and automatically appending a "," if so before
+// writing the next field.
+type Stream struct {
+	*jsoniter.Stream
+	wmTracker []bool
```

Comment:
I'd like a comment on `wmTracker`. Why is it a slice? You seem to always use the last element. More comments please :)